routers/web/repo/treelist.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"net/http"
+
+	"code.gitea.io/gitea/modules/base"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/git"
+
+	"github.com/go-enry/go-enry/v2"
+)
+
+// TreeList get all files' entries of a repository
+func TreeList(ctx *context.Context) {
+	tree, err := ctx.Repo.Commit.SubTree("/")
+	if err != nil {
+		ctx.ServerError("Repo.Commit.SubTree", err)
+		return
+	}
+
+	entries, err := tree.ListEntriesRecursive()
+	if err != nil {
+		ctx.ServerError("Repo.Commit.SubTree", err)
+		return
+	}
+	entries.CustomSort(base.NaturalSortLess)
+
+	var files []string
+	for _, entry := range entries {
+		if !isExcludedEntry(entry) {
+			files = append(files, entry.Name())
+		}
+	}
+	ctx.JSON(http.StatusOK, files)
+}
+
+func isExcludedEntry(entry *git.TreeEntry) bool {
+	if entry.IsDir() {
+		return true
+	}
+
+	if entry.IsSubModule() {
+		return true
+	}
+
+	if enry.IsVendor(entry.Name()) {
+		return true
+	}
+
+	return false
```

Comment:
```suggestion
	return entry.IsDir() || entry.IsSubModule() || enry.IsVendor(entry.Name())
```
Why the IsVendor check?
Should vendored files not be searchable?

Reply:
I think it makes sense to skip vendored files, like we also do that in search

Reply:
See the comment of  KN4CK3R  https://github.com/go-gitea/gitea/pull/15028#issuecomment-802723331

I do not want to shorten the code too aggressively, these `if` are clear and extensible.

Reply:
Okay, if you want to keep the `if`s...
I'd prefer it shortened, but I can live with the current approach as well.

Reply:
Personally I would also prefer shortened